scalding-jdbc-macros/src/main/scala/com/twitter/scalding/jdbc/macros/impl/JdbcMacroImpl.scala

```diff
@@ -0,0 +1,117 @@
+
+    if (!IsCaseClassImpl.isCaseClassType(c)(T.tpe))
+      c.abort(c.enclosingPosition, s"""We cannot enforce ${T.tpe} is a case class, either it is not a case class or this macro call is possibly enclosed in a class.
+        This will mean the macro is operating on a non-resolved type.""")
+
+    val defaultArgs = getDefaultArgs[T](c)(T)
+
+    // Field To JDBCColumn
+    def matchField(oTpe: Type,
+      fieldName: FieldName,
+      defaultValOpt: Option[c.Expr[String]],
+      annotationInfo: List[(Type, Option[Int])],
+      nullable: Boolean): scala.util.Try[c.Expr[ColumnDefinition]] = {
+
+      oTpe match {
+        // String handling
+        case tpe if tpe =:= typeOf[String] => StringTypeHandler(c)(fieldName, defaultValOpt, annotationInfo, nullable)
+        case tpe if tpe =:= typeOf[Short] => NumericTypeHandler(c)(fieldName, defaultValOpt, annotationInfo, nullable, "SMALLINT")
+        case tpe if tpe =:= typeOf[Int] => NumericTypeHandler(c)(fieldName, defaultValOpt, annotationInfo, nullable, "INT")
+        case tpe if tpe =:= typeOf[Long] => NumericTypeHandler(c)(fieldName, defaultValOpt, annotationInfo, nullable, "BIGINT")
+        case tpe if tpe =:= typeOf[Double] => NumericTypeHandler(c)(fieldName, defaultValOpt, annotationInfo, nullable, "DOUBLE")
+        case tpe if tpe =:= typeOf[Boolean] => NumericTypeHandler(c)(fieldName, defaultValOpt, annotationInfo, nullable, "TINYINT")
+        case tpe if tpe =:= typeOf[scala.math.BigInt] => NumericTypeHandler(c)(fieldName, defaultValOpt, annotationInfo, nullable, "BIGINT")
+        case tpe if tpe =:= typeOf[java.util.Date] => DateTypeHandler(c)(fieldName, defaultValOpt, annotationInfo, nullable)
+
+        case tpe if tpe.erasure =:= typeOf[Option[Any]] =>
+          if (defaultValOpt.isDefined)
+            Failure(new Exception(s"Case class ${T.tpe} has field ${fieldName}: ${oTpe.toString}, with a default value. Options cannot have default values"))
+          else
+            matchField(tpe.asInstanceOf[TypeRefApi].args.head, fieldName, None, annotationInfo, true)
+
+        // default
+        case _ => Failure(new Exception(s"Case class ${T.tpe} has field ${fieldName}: ${oTpe.toString}, which is not supported for talking to JDBC"))
+      }
+    }
+
+    def expandMethod(outerTpe: Type): Iterable[c.Expr[ColumnDefinition]] = {
+      outerTpe
+        .declarations
+        .collect { case m: MethodSymbol if m.isCaseAccessor => m }
+        .map { m =>
+          val fieldName = m.name.toTermName.toString
+          val defaultVal = defaultArgs.get(fieldName)
+          val annotationInfo: List[(Type, Option[Int])] = m.annotations
+            .map(t => (t.tpe, t.scalaArgs))
+            .collect {
+              case (tpe, List(Literal(Constant(siz: Int)))) if tpe =:= typeOf[com.twitter.scalding.jdbc.macros.size] => (tpe, Some(siz))
+              case (tpe, _) if tpe <:< typeOf[com.twitter.scalding.jdbc.macros.ScaldingJdbcAnnotation] => (tpe, None)
```

Comment:
If a size annotation is above, but the size is a reference to a number elsewhere I don't believe it will be a literal constant so we will lose that value.